presto-main/src/main/java/io/prestosql/type/TypeUtils.java

```diff
@@ -45,6 +47,15 @@ public static int expectedValueSize(Type type, int defaultSize)
         if (type instanceof FixedWidthType) {
             return ((FixedWidthType) type).getFixedSize();
         }
+        else if (type instanceof VarcharType) {
+            // If bound on length of varchar is smaller than defaultSize, use that as expected size
+            return ((VarcharType) type).getLength()
+                    .map(length -> Math.min(length, defaultSize))
```

Comment:
Varchars usually won't occupy full declared length.

Reply:
`varchar` length is "character count", and`varchar` is encoded in UTF-8 which means up to 4 bytes per character.  If this is just an expected size, assuming ascii is reasoanble, but we should note that in a comment here.

Reply:
Added a comment for UTF-8

Reply:
I still don't think that assuming that varchar will occupy entire length is the correct one. It seems very pessimistic. What do you think @dain?

Reply:
@sopel39 that's why we do `min` here, right?

Reply:
I'm not sure what contract of `defaultSize` is here. Could it be excessively large?

Reply:
In current usage `defaultSize` is 16, 32 or 100 depending on where it's getting called from. This change should just reduce the expected size estimate when a smaller bound is known (E.g. varchar(10)). This would help reduce memory usage in cases where a large no. of TypeSet are generated but each set has small no. of entries which are a few characters long.

Reply:
I think we should assume the default size is one byte.  I would update the comment a bit:
```
It can take up to 4 bytes per character due to UTF-8 encoding, but we assume the data is ASCII and only needs one byte.
```

Reply:
Updated the comments in code